app/actions/favorites.js

```diff
@@ -16,9 +17,13 @@ export function readFavorites() {
 
 export function addFavoriteTrack(track) {
   const favorites = store.get('favorites');
-  favorites.tracks = _.concat(favorites.tracks, track);
+  const newTrack = {
+    ...track,
+    uuid: uuidv4()
```

Comment:
I've noticed the track is removed from favorites by `uuid` but usually when it's added it's missing that identifier. I've decided to add it with each `addFavoriteTrack` action as it helped me in converting best track item to a favorite one.

Reply:
Uuids for favorite tracks were not generated, because they used to be added only from the queue, and the queue adds uuids to its items when they're added to it:

https://github.com/nukeop/nuclear/blob/4566bc2ae6f57a1200e07398e045a22c857a0350/app/actions/queue.js#L19

Now that you're implementing adding favorite tracks from other sources, I suppose this will be necessary. What is important to keep in mind is to clone the track on adding so that we do not mutate the state and mess with the queue's ids, so what you did is a good way to handle this.